packages/vm/src/memory.rs

```diff
@@ -25,6 +25,35 @@ pub struct Region {
     /// The number of bytes used in this region
     pub length: u32,
 }
+type RegionBytes = [u8; size_of::<Region>()];
+
+impl Region {
+    fn from_wasm_bytes(bytes: RegionBytes) -> Self {
+        let offset = u32::from_le_bytes(bytes[0..4].try_into().unwrap());
```

Comment:
Not because of `#[repr(C)]`, but wasm itself is little endian, so they are LE in the wasm memory.

Agree with the suggestion 👍

Reply:
okay but how did we know before that the `struct Region` in cosmwasm-std and `struct Region` in cosmwasm-vm used the same endianness? Pure luck?

Reply:
Oh, good point. I guess it would have produced a wrong `Region` on a big endian architecture. Those are pretty rare though: x86 is LE, some ARM chips supports both, but most default to LE. I believe the last wide-spread use was PowerPC in game consoles like the Wii

Reply:
`#[repr(C)]` does not add any guarantees here? Can we somehow assert that the target of cosmwasm-vm is LE?

Reply:
I think `#[repr(C)]` gives guarantees about order, size and alignment of fields, but not endianness. That is target specific.
We could do something like:
```
#[cfg(target_endian = "big")]
compile_error!("big endian systems are not supported");
```

Reply:
I added the code with an explanatory comment.
Technically the code in this PR should also work on a big endian architecture, but I wouldn't vouch for wasmer supporting it and it's also good to have the error here, since we want to switch back to the direct pointer access at some point.